compiler/damlc/daml-stdlib-src/DA/Internal/Interface.daml

```diff
@@ -91,21 +91,45 @@ toInterfaceContractId : forall i t. HasToInterface t i => ContractId t -> Contra
 toInterfaceContractId = coerceContractId
 
 -- | (1.dev only) Convert an interface contract id into a template
--- contract id. In order to verify that the underlying contract has
--- the expected template type, this will perform a fetch. For example:
+-- contract id. For example, `fromInterfaceContractId @MyTemplate cid`.
 --
+-- This function does not verify that the interface contract id
+-- actually points to a template of the given type. This means
+-- that a subsequent `fetch`, `exercise`, or `archive` may fail, if,
```

Comment:
I think we should have a few `submitMustFail` style tests for this statement here